includes/functions/functions_general.php

```diff
@@ -208,8 +208,13 @@ function zen_get_buy_now_button($product_id, string $buy_now_link, $additional_l
             global $template;
             $image = BUTTON_IMAGE_SOLD_OUT; 
             $alt = BUTTON_SOLD_OUT_ALT; 
-            $return_button = '<span class="text-center">' . zen_image($template->get_template_dir($image, DIR_WS_TEMPLATE, $current_page_base, 'buttons/' . $_SESSION['language'] . '/') . $image, $alt, '', '', '') . '</span>'; 
+            if (strtolower(IMAGE_USE_CSS_BUTTONS) == 'yes') {
+                $return_button =  zenCssButton($image, $alt, 'button', '', 'id=soldOutButton');
```

Comment:
An `id=` is going to generate invalid HTML if multiple products are 'sold out' on a products' listing page.  Suggest using `class=` instead.

Reply:
Good spot thanks.
Cannot use class as html_output line 451 will create a duplicate class.
` $css_button = '<input class="' . $mouse_out_class . '" ' . $css_button_js . ' type="submit" value="' . $text . '"' . $tooltip . $parameters . '>';`

I suggest the i remove the parameter. 
The button is identified in css by `span.button_sold_out`

Reply:
Hold off; this should be using
```php
$return_button = zen_image_button($image, $alt);
```
That will generate a `button` with a class of `button_sold_out`.

Reply:
... and the comparison should be strict, i.e. `=== 'yes'`.

Reply:
I tried zen_image_button that creates a clickable button that takes you to the cart. 
That was why I used zenCssButton. 
ZenCssButton also generate a button with class button_sold_out. Clicking it has no action, the same as the image

Reply:
I'm not seeing that generation of a 'clickable' button.  When I make the change that I suggested (less the CSS updates), I'm presented with an unclickable `span` that says "Sold Out"; here's the hover image
![image](https://github.com/zencart/zencart/assets/2685585/fcdf115c-fd29-428e-a444-a999f3184b1e)

Reply:
OK I'll try it again. That is where I started. Ignore last push. I'll check it out and push again if ok.

Reply:
Why should I doubt the master. Works a treat!